packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-extra-settings.tsx

```diff
@@ -105,11 +113,18 @@ export const PipelineExtraSettings: React.FunctionComponent<
 
 const mapState = ({
   autoPreview,
-  pipelineBuilder: { pipelineMode },
-}: RootState) => ({
-  isAutoPreview: autoPreview,
-  pipelineMode,
-});
+  pipelineBuilder: { pipelineMode, textEditor, stageEditor },
+}: RootState) => {
+  const isPipelineModeDisabled =
+    pipelineMode === 'as-text'
+      ? textEditor.syntaxErrors.length > 0
+      : stageEditor.stages.some((x) => x.syntaxError);
```

Comment:
Should we use builder helper here? Is the concern that we don't want to disable on network errors? If that's the case it might be a good idea to not account for them anywhere, not only here and so I think that using a helper is still warranted

Reply:
Yes, used `getIsPipelineInvalidFromBuilderState` to do that. I extended it to ignore the server errors, because the toggle should only be disabled if there's a parse error.